openscap_daemon/system.py

```diff
@@ -640,6 +640,17 @@ def generate_report_for_task_result(self, task_id, result_id):
             self.config
         )
 
+    def generate_fix_for_task_result(self, task_id, result_id, fix_type):
+        task = None
+        with self.tasks_lock:
+            task = self.tasks[task_id]
+
+        return task.generate_fix_for_result(
```

Comment:
shouldn't this be in the tasks_lock? can you unlock the task for changes and then generate a fix from it?